### PR TITLE
Disable removing system-catalog

### PR DESCRIPTION
**Pull request by rmweir**
Problem:
Prior, system-catalog was hidden to prevent these actions. This prevented user from viewing errors with system-catalog. Prior to that, user was able to delete system-catalog which could be dangerous.

Solution:
Allow user to view system catalog, but prevent them from deleting it.

Issue:
https://github.com/rancher/rancher/issues/22771

Related Issue:
https://github.com/rancher/rancher/issues/22490

### CODE REVIEW

File: pkg/api/customization/catalog/store.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	}
+}
+
+func (s *Store) Delete(apiContext *types.APIContext, schema *types.Schema, id string) (map[string]interface{}, error) {
+	catalog, err := s.ByID(apiContext, schema, id)
+	if err != nil {
+		return nil, err
+	}
+	if catalog["url"] == systemLibraryURL && catalog["name"] == "system-library" {
+		return nil, httperror.NewAPIError(httperror.Conflict, fmt.Sprint("[catalog store] Unathorized to delete system-library catalog"))
```
Comment (dramich) [inline pkg/api/customization/catalog/store.go]: The error should be a `InvalidBodyContent`, `conflict` would most likely apply to a PUT request. The message itself should be updated to `not allowed to delete system-library`

File: pkg/api/customization/catalog/catalog.go [hunk 2]
```diff
@@ -18,9 +18,16 @@ import (
...
 )
 
+const (
+	systemLibraryURL = "https://git.rancher.io/system-charts"
+)
+
 func Formatter(apiContext *types.APIContext, resource *types.RawResource) {
 	resource.AddAction(apiContext, "refresh")
 	resource.Links["exportYaml"] = apiContext.URLBuilder.Link("exportYaml", resource)
+	if resource.Values["url"] == systemLibraryURL && resource.Values["name"] == "system-library" {
```
Comment (dramich) [inline pkg/api/customization/catalog/catalog.go]: Turn `"system-library" ` into a const as well.

File: pkg/api/customization/catalog/catalog.go [hunk 3]
```diff
@@ -15,12 +15,36 @@ import (
...
 
-func Formatter(apiContext *types.APIContext, resource *types.RawResource) {
+const (
+	SystemLibraryURL            = "https://git.rancher.io/system-charts"
+	SystemCatalogName           = "system-library"
+	embededSystemCatalogSetting = "system-catalog"
+)
+
+type Wrapper struct {
+	SettingLister v3.SettingLister
```
Comment (dramich) [inline pkg/api/customization/catalog/catalog.go]: Use the settings package for this.

File: pkg/api/customization/catalog/validator.go [hunk 4]
```diff
@@ -15,7 +16,17 @@ var (
 	controlEncoded = regexp.MustCompile("%[0-1][0-9,a-f,A-F]")
 )
 
-func Validator(request *types.APIContext, schema *types.Schema, data map[string]interface{}) error {
+func (w Wrapper) Validator(request *types.APIContext, schema *types.Schema, data map[string]interface{}) error {
+	if request.Method == http.MethodPut {
```
Comment (dramich) [inline pkg/api/customization/catalog/validator.go]: Let's move this into a store update method then use a helper so this logic isn't duplicated

**Comment by riaan53** [comment]
Will we still be able to edit the chart for air gap deployments? Currently we just delete and add the system-chart (in automated way) again with new values.

**Comment by dramich** [comment]
@riaan53 we have bundled in the system-charts for air gap, so providing a flag to rancher will use that version vs having to get something external. This change does lock down system-charts so it can't be deleted. https://github.com/rancher/rancher/issues/20387

**Comment by dramich** [comment]
lgtm

**Pull request merged**